app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -55,24 +55,42 @@ def validate_authentication_status
       {:available => true, :message => nil}
     end
 
+    def self.verify_ssl_mode(security_protocol)
+      case security_protocol
+      when nil, '', 'non-ssl', 'ssl-without-validation'
+        OpenSSL::SSL::VERIFY_NONE
+      else
+        OpenSSL::SSL::VERIFY_PEER
+      end
+    end
+
+    def self.entrypoint(host, port, security_protocol)
+      return URI::HTTP.build(:host => host, :port => port.to_i).to_s if security_protocol == 'non-ssl'
```

Comment:
I think this should also use HTTP when security_protocol is missing (nil, '')? See also comment in `verify_ssl_mode()`.
I suspect this is why some of your tests failed on missing https:// VCR where they previously worked with http://.